pkgs/top-level/all-packages.nix

```diff
@@ -2601,15 +2601,47 @@ with pkgs;
 
   go-audit = callPackage ../tools/system/go-audit { };
 
-  gopass = callPackage ../tools/security/gopass { };
+  gopass = callPackage ../tools/security/gopass {
+    # pinned due to https://github.com/gopasspw/gopass/issues/2280
+    # when go_1_18 is bumped to 1.18.5 please remove __buildGo1183Module
+    buildGoModule = __buildGo1183Module;
+  };
+
+  gopass-hibp = callPackage ../tools/security/gopass/hibp.nix {
+    # pinned due to https://github.com/gopasspw/gopass/issues/2280
+    # when go_1_18 is bumped to 1.18.5 please remove __buildGo1183Module
+    buildGoModule = __buildGo1183Module;
+  };
 
-  gopass-hibp = callPackage ../tools/security/gopass/hibp.nix { };
+  gopass-jsonapi = callPackage ../tools/security/gopass/jsonapi.nix {
+    # pinned due to https://github.com/gopasspw/gopass/issues/2280
+    # when go_1_18 is bumped to 1.18.5 please remove __buildGo1183Module
+    buildGoModule = __buildGo1183Module;
+  };
 
-  gopass-jsonapi = callPackage ../tools/security/gopass/jsonapi.nix { };
+  git-credential-gopass = callPackage ../tools/security/gopass/git-credential.nix {
+    # pinned due to https://github.com/gopasspw/gopass/issues/2280
+    # when go_1_18 is bumped to 1.18.5 please remove __buildGo1183Module
+    buildGoModule = __buildGo1183Module;
+  };
 
-  git-credential-gopass = callPackage ../tools/security/gopass/git-credential.nix { };
+  gopass-summon-provider = callPackage ../tools/security/gopass/summon.nix {
+    # pinned due to https://github.com/gopasspw/gopass/issues/2280
+    # when go_1_18 is bumped to 1.18.5 please remove __buildGo1183Module
+    buildGoModule = __buildGo1183Module;
+  };
 
-  gopass-summon-provider = callPackage ../tools/security/gopass/summon.nix { };
+  # go 1.18 requires a newer Apple SDK to be build. See commit message for more details.
+  __buildGo1183Module = darwin.apple_sdk_11_0.callPackage ../development/go-modules/generic {
+    go = buildPackages.go_1_18.overrideAttrs (_: rec {
+      version = "1.18.3";
```

Comment:
Would it not be easier to just build with go 1.17 rather than building a complete go compiler for one program?

Reply:
I think gopass needs go 1.18, due to the use of generics.

With the strategy from https://github.com/NixOS/nixpkgs/pull/182784#issuecomment-1193659652, we'll end up shipping the fix in go 1.18, before 1.18.5 is released.

Reply:
Nevermind. Gopass does not build with go 1.17.